L1Trigger/TrackFindingTracklet/src/TrackletConfigBuilder.cc

```diff
@@ -53,18 +58,130 @@ TrackletConfigBuilder::TrackletConfigBuilder(const Settings& settings) : setting
+        const float phiOffsetHybrid = 0.5 * dphisectorHG_;
+        phiMin += phiOffsetHybrid;
+        phiMax += phiOffsetHybrid;
+        if (dtcPhiRange.find(layer) == dtcPhiRange.end()) {
+          dtcPhiRange[layer] = {phiMin, phiMax};
+        } else {
+          dtcPhiRange.at(layer).first  = std::min(phiMin, dtcPhiRange.at(layer).first);
+          dtcPhiRange.at(layer).second = std::max(phiMax, dtcPhiRange.at(layer).second);
+        }
+      }
+    }
+    for (const auto& p : dtcPhiRange) {
+      const unsigned int numSlots = setup->numATCASlots();
+      std::string dtcName = settings_.slotToDTCname(dtcId%numSlots);
+      if (dtcId >= numSlots) dtcName = "neg" + dtcName;
+      DTCinfo info;
+      info.name = dtcName;
+      info.layer = p.first;
+      info.phimin = p.second.first;
+      info.phimax = p.second.second;
+      vecDTCinfo_unsorted.push_back(info);
+    }
+  }
+
+  // Put DTCinfo vector in traditional order (PS first). (Needed?)
+  for (const DTCinfo& info : vecDTCinfo_unsorted) {
+    string dtcname = info.name;
+    if (dtcname.find("PS") != std::string::npos) {
+      vecDTCinfo_.push_back(info);
+    }
+  }
+  for (const DTCinfo& info : vecDTCinfo_unsorted) {
+    string dtcname = info.name;
+    if (dtcname.find("PS") == std::string::npos) {
+      vecDTCinfo_.push_back(info);
+    }
+  }
+}
+
+//--- Write DTC phi ranges to file to support stand-alone emulation.
+//--- (Only needed to support stand-alone emulation)
+
+void TrackletConfigBuilder::writeDTCphirange() const {
+  bool first = true;
+  for (const DTCinfo& info : vecDTCinfo_) {
+    string dirName = settings_.tablePath();
+    string fileName = dirName + "../dtcphirange.dat";
+    std::ofstream out;
+    openfile(out, first, dirName, fileName, __FILE__, __LINE__);
+    if (first) {
+      out<<"// phi ranges of modules read by each DTC"<<endl;
+      out<<"// (Used by stand-alone emulation)"<<endl;
+    }
+    out<<"vecDTCinfo_.push_back( {"""<<info.name<<""", "<<info.layer<<", "<<info.phimin<<", "<<info.phimax<<"} );"<<endl;
+    out.close();
+    first = false;
+  }
+}
+
+#else
+
+void TrackletConfigBuilder::setDTCphirange(const tt::Setup* setup) {
+#include "L1Trigger/TrackFindingTracklet/data/dtcphirange.txt"
+}
+
+#endif
+
+//--- Helper fcn. to get the layers/disks for a seed
+
 std::pair<unsigned int, unsigned int> TrackletConfigBuilder::seedLayers(unsigned int iSeed) {
   return std::pair<unsigned int, unsigned int>(settings_.seedlayers(0, iSeed), settings_.seedlayers(1, iSeed));
 }
 
+//--- Method to initialize the regions and VM in each layer
+
 void TrackletConfigBuilder::initGeom() {
   for (unsigned int ilayer = 0; ilayer < N_LAYER + N_DISK; ilayer++) {
     double dphi = dphisectorHG_ / NRegions_[ilayer];
     for (unsigned int iReg = 0; iReg < NRegions_[ilayer]; iReg++) {
       std::vector<std::pair<unsigned int, unsigned int> > emptyVec;
       projections_[ilayer].push_back(emptyVec);
+      // FIX: sector doesn't have hourglass shape
```

Comment:
Not sure what you want to fix here. The projections are organized by just sorting on the top bits of the phi position.

Reply:
I believe some of these regions (e.g. those with layer radius near rcrit) may lie outside the hourglass overlap shape? (The code in initGeom() doesn't take into account the radial dependence of the hourglass shape). Though I don't know if this leads to any waste of resources. May be silly idea on my part ...